docs/powergate/cidconfig.md

```diff
@@ -0,0 +1,120 @@
+# Managing Storage with the CidConfig
+
+Every [FFS instance](ffs.md) can manage how data is stored on IPFS and Filecoin using the **CidConfig** ([details below](#cidconfig-details)). The CidConfig is a powerful tool to customize all the details of how you store data on Filecoin, make it available over IPFS, enforce replication, manage expiring deals, and more.
+
+## Setting the CidConfig
+
+In every Powergate deployment there are three ways to manage CidConfigs throughout the system.
+
+1. The FFS instance default CidConfig. This is initially set by the system default CidConfig. It can be modified by the FFS instance owner after creation.
+2. The storage request CidConfig. This will use the FFS instance default, but a custom CidConfig can also be supplied at request time.
+3. A storage update CidConfig. Any CidConfigs attached to existing stored data can be updated with a new CidConfig. The FFS instance will then work to modify the way data is stored to match the new configuration
+
+### Get the default CidConfig of an FFS instance
+
+View the current default CidConfig of an FFS instance. 
+
+```bash
+pow ffs config default -t <token>
+```
+
+### Set the default CidConfig of an FFS instance
+
+To set the default config to one stored in `new-config.json`.
+
+```bash
+pow ffs config set new-config.json -t <token>
+```
+
+### Set a custom CidConfig at storage time
+
+To make a storage request for a CID with a custom config stored in `custom-config.json`. If you leave the config option off of the following command, the default CidConfig would be used.
+
+```bash
+pow ffs push <cid> -t <token> -c custom-config.json
+```
+
+### Get the CidConfig of previously stored data
+
+To pull the CidConfig associated with data already managed by the FFS instance, use the _<cid>_ of the stored data.
+
+```bash
+pow ffs pull <cid> -t <token>
+```
+
+### Update the CidConfig of existing data
+
+To update the CidConfig of data already stored and managed by the Powergate with a new config stored in `updated-config.json`.
+
+```bash
+pow ffs push <cid> -t <token> -c updated-config.json
```

Comment:
This should be:
`pow ffs push <cid> -t <token> -o -c updated-config.json`

The `-o` flag stands for _override_. This is required anytime a newer version of an existing CidConfig is provided for a Cid. The rationale about this is to provide a safe way to replace cidconfigs and distinguishing the action from the creation.

If the above command is run, it will say that _there's already a cid config for that Cid, use the override flag_, to be sure the intention of the user is really updating an already stored thing.

It has no other intention than trying to be sure that what will happen (replacing how an already stored thing is stored) was the original intention rather than "uhm, i'll store this thing for the first time". Mentioning this since this could be taken out and not explicitly try to distinguish replacing than pushing for first time.